sklearn/preprocessing/data.py

```diff
@@ -2028,7 +2028,11 @@ def fit(self, X, y=None):
 
         if y is not None:
             for i in range(len_data):
-                X_key = tuple(X.take(i, axis=0).take(inclusion_used).flatten())
+                X_key_1 = X.take(i, axis=0)
```

Comment:
Why were you doing `X.take(i, axis=0)` anyway? I think `X[i]` should work fine; as should `for x in X`